hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/handlers/EthereumTransactionHandler.java

```diff
@@ -105,16 +110,29 @@ public void preHandle(@NonNull final PreHandleContext context) throws PreCheckEx
 
     @Override
     public void pureChecks(@NonNull final TransactionBody txn) throws PreCheckException {
-        final var ethTxData = populateEthTxData(
-                requireNonNull(txn.ethereumTransactionOrThrow().ethereumData()).toByteArray());
-        validateTruePreCheck(nonNull(ethTxData), INVALID_ETHEREUM_TRANSACTION);
-        final byte[] callData = ethTxData.hasCallData() ? ethTxData.callData() : new byte[0];
-        final var intrinsicGas =
-                gasCalculator.transactionIntrinsicGasCost(org.apache.tuweni.bytes.Bytes.wrap(callData), false);
-        validateTruePreCheck(ethTxData.gasLimit() >= intrinsicGas, INSUFFICIENT_GAS);
-        // Do not allow sending HBars to Burn Address
-        if (ethTxData.value().compareTo(BigInteger.ZERO) > 0) {
-            validateFalsePreCheck(Arrays.equals(ethTxData.to(), EMPTY_ADDRESS), INVALID_SOLIDITY_ADDRESS);
+        try {
+            final var ethTxData = populateEthTxData(
+                    requireNonNull(txn.ethereumTransactionOrThrow().ethereumData())
+                            .toByteArray());
+            validateTruePreCheck(nonNull(ethTxData), INVALID_ETHEREUM_TRANSACTION);
+            final byte[] callData = ethTxData.hasCallData() ? ethTxData.callData() : new byte[0];
+            final var intrinsicGas =
+                    gasCalculator.transactionIntrinsicGasCost(org.apache.tuweni.bytes.Bytes.wrap(callData), false);
+            validateTruePreCheck(ethTxData.gasLimit() >= intrinsicGas, INSUFFICIENT_GAS);
+            // Do not allow sending HBars to Burn Address
+            if (ethTxData.value().compareTo(BigInteger.ZERO) > 0) {
+                validateFalsePreCheck(Arrays.equals(ethTxData.to(), EMPTY_ADDRESS), INVALID_SOLIDITY_ADDRESS);
+            }
+        } catch (@NonNull final Exception e) {
+            final var contractMetrics = component.contractMetrics();
+            contractMetrics.incrementRejectedTx(CONTRACT_CALL);
```

Comment:
Did you want `EthereumTransaction` here?  Also it seems that these lines repeat in the handler classes.  Should they be moved to a utility method?

Reply:
First: Yes, TY!
Second: Could do that, but where?  Let me think. It's the dependency on the `ContractServiceComponent` - it'll have to be passed in.

Reply:
But will wait for follow-on PR with P2 metrics ....

Reply:
NVM, did it in this PR.  Abstracted commonality into a new `AbstractContractTransactionHandler` class to sit between the contract handlers and `TransactionHandler`.

Didn't add unit tests for this new class because current tests (for the existing handlers) cover the functionality.  Unless you disagree?